src/transport/raw/PeerAddress.h

```diff
@@ -133,18 +136,21 @@ class PeerAddress
     {
         char ip_addr[kInetMaxAddrLen];
 
+        char interface[Inet::InterfaceId::kMaxIfNameLength + 1];
+        CHIP_ERROR err = mInterface.GetInterfaceName(interface, sizeof(interface));
+
         switch (mTransportType)
         {
         case Type::kUndefined:
             snprintf(buf, bufSize, "UNDEFINED");
             break;
         case Type::kUdp:
             mIPAddress.ToString(ip_addr);
-            snprintf(buf, bufSize, "UDP:%s:%d", ip_addr, mPort);
+            snprintf(buf, bufSize, "UDP:[%s]:%d%%%s", ip_addr, mPort, err == CHIP_NO_ERROR ? interface : "(err)");
```

Comment:
Please do this conditional in above rather than copy and paste and place the result in `interface`

Also need to handle null interface: you should not have `[abc%]:123` when interface name is empty.